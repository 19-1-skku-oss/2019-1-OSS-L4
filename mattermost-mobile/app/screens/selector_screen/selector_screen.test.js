// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';
import {IntlProvider} from 'react-intl';

import Preferences from 'mattermost-redux/constants/preferences';

import SelectorScreen from './selector_screen.js';

jest.mock('rn-fetch-blob', () => ({
    fs: {
        dirs: {
            DocumentDir: () => jest.fn(),
            CacheDir: () => jest.fn(),
        },
    },
}));

jest.mock('rn-fetch-blob/fs', () => ({
    dirs: {
        DocumentDir: () => jest.fn(),
        CacheDir: () => jest.fn(),
    },
}));

const user1 = {id: 'id', username: 'username'};
const user2 = {id: 'id2', username: 'username2'};

const getProfiles = async () => {
    return {
        data: [user1, user2],
        error: {},
    };
};

const searchProfiles = async () => {
    return {
        data: [user2],
        error: {},
    };
};

const channel1 = {id: 'id', name: 'name', display_name: 'display_name'};
const channel2 = {id: 'id2', name: 'name2', display_name: 'display_name2'};

const getChannels = async () => {
    return {
        data: [channel1, channel2],
        error: {},
    };
};

const searchChannels = async () => {
    return {
        data: [channel2],
        error: {},
    };
};

const intlProvider = new IntlProvider({locale: 'en'}, {});
const {intl} = intlProvider.getChildContext();

describe('SelectorScreen', () => {
    const actions = {
        getProfiles,
        getChannels,
        searchProfiles,
        searchChannels,
    };

    const baseProps = {
        actions,
        currentTeamId: 'someId',
        navigator: {
            setOnNavigatorEvent: jest.fn(),
        },
        onSelect: jest.fn(),
        data: [{text: 'text', value: 'value'}],
        dataSource: null,
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot for explicit options', async () => {
        const wrapper = shallow(
            <SelectorScreen {...baseProps}/>,
            {context: {intl}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for users', async () => {
        const props = {
            ...baseProps,
            dataSource: 'users',
            data: [user1, user2],
        };

        const wrapper = shallow(
            <SelectorScreen {...props}/>,
            {context: {intl}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        wrapper.setState({isLoading: false});
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for channels', async () => {
        const props = {
            ...baseProps,
            dataSource: 'channels',
            data: [channel1, channel2],
        };

        const wrapper = shallow(
            <SelectorScreen {...props}/>,
            {context: {intl}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        wrapper.setState({isLoading: false});
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for searching', async () => {
        const props = {
            ...baseProps,
            dataSource: 'channels',
            data: [channel1, channel2],
        };

        const wrapper = shallow(
            <SelectorScreen {...props}/>,
            {context: {intl}},
        );
        wrapper.setState({isLoading: false, searching: true, term: 'name2'});
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
