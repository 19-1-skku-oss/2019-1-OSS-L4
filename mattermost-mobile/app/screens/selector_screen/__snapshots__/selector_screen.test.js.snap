// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectorScreen should match snapshot for channels 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "marginVertical": 5,
      }
    }
  >
    <SearchBarIos
      autoCapitalize="none"
      backgroundColor="transparent"
      blurOnSubmit={true}
      cancelTitle="Cancel"
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(61,60,64,0.2)",
          "color": "#3d3c40",
          "fontSize": 15,
        }
      }
      leftComponent={null}
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(61,60,64,0.5)"
      searchIconCollapsedMargin={10}
      searchIconExpandedMargin={10}
      tintColorDelete="rgba(61,60,64,0.5)"
      tintColorSearch="rgba(61,60,64,0.5)"
      titleCancelColor="#3d3c40"
      value=""
    />
  </View>
  <CustomList
    data={Array []}
    listType="flat"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</View>
`;

exports[`SelectorScreen should match snapshot for channels 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "marginVertical": 5,
      }
    }
  >
    <SearchBarIos
      autoCapitalize="none"
      backgroundColor="transparent"
      blurOnSubmit={true}
      cancelTitle="Cancel"
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(61,60,64,0.2)",
          "color": "#3d3c40",
          "fontSize": 15,
        }
      }
      leftComponent={null}
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(61,60,64,0.5)"
      searchIconCollapsedMargin={10}
      searchIconExpandedMargin={10}
      tintColorDelete="rgba(61,60,64,0.5)"
      tintColorSearch="rgba(61,60,64,0.5)"
      titleCancelColor="#3d3c40"
      value=""
    />
  </View>
  <CustomList
    data={Array []}
    listType="flat"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</View>
`;

exports[`SelectorScreen should match snapshot for explicit options 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "marginVertical": 5,
      }
    }
  >
    <SearchBarIos
      autoCapitalize="none"
      backgroundColor="transparent"
      blurOnSubmit={true}
      cancelTitle="Cancel"
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(61,60,64,0.2)",
          "color": "#3d3c40",
          "fontSize": 15,
        }
      }
      leftComponent={null}
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(61,60,64,0.5)"
      searchIconCollapsedMargin={10}
      searchIconExpandedMargin={10}
      tintColorDelete="rgba(61,60,64,0.5)"
      tintColorSearch="rgba(61,60,64,0.5)"
      titleCancelColor="#3d3c40"
      value=""
    />
  </View>
  <CustomList
    data={
      Array [
        Object {
          "text": "text",
          "value": "value",
        },
      ]
    }
    listType="flat"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</View>
`;

exports[`SelectorScreen should match snapshot for searching 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "marginVertical": 5,
      }
    }
  >
    <SearchBarIos
      autoCapitalize="none"
      backgroundColor="transparent"
      blurOnSubmit={true}
      cancelTitle="Cancel"
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(61,60,64,0.2)",
          "color": "#3d3c40",
          "fontSize": 15,
        }
      }
      leftComponent={null}
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(61,60,64,0.5)"
      searchIconCollapsedMargin={10}
      searchIconExpandedMargin={10}
      tintColorDelete="rgba(61,60,64,0.5)"
      tintColorSearch="rgba(61,60,64,0.5)"
      titleCancelColor="#3d3c40"
      value="name2"
    />
  </View>
  <CustomList
    data={Array []}
    listType="flat"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</View>
`;

exports[`SelectorScreen should match snapshot for users 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "marginVertical": 5,
      }
    }
  >
    <SearchBarIos
      autoCapitalize="none"
      backgroundColor="transparent"
      blurOnSubmit={true}
      cancelTitle="Cancel"
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(61,60,64,0.2)",
          "color": "#3d3c40",
          "fontSize": 15,
        }
      }
      leftComponent={null}
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(61,60,64,0.5)"
      searchIconCollapsedMargin={10}
      searchIconExpandedMargin={10}
      tintColorDelete="rgba(61,60,64,0.5)"
      tintColorSearch="rgba(61,60,64,0.5)"
      titleCancelColor="#3d3c40"
      value=""
    />
  </View>
  <CustomList
    data={Array []}
    listType="section"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</View>
`;

exports[`SelectorScreen should match snapshot for users 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "marginVertical": 5,
      }
    }
  >
    <SearchBarIos
      autoCapitalize="none"
      backgroundColor="transparent"
      blurOnSubmit={true}
      cancelTitle="Cancel"
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(61,60,64,0.2)",
          "color": "#3d3c40",
          "fontSize": 15,
        }
      }
      leftComponent={null}
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(61,60,64,0.5)"
      searchIconCollapsedMargin={10}
      searchIconExpandedMargin={10}
      tintColorDelete="rgba(61,60,64,0.5)"
      tintColorSearch="rgba(61,60,64,0.5)"
      titleCancelColor="#3d3c40"
      value=""
    />
  </View>
  <CustomList
    data={Array []}
    listType="section"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</View>
`;
